.github/workflows/functional-test-noncloud.yaml

```diff
@@ -0,0 +1,445 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------
+
+name: Functional Tests (with Non-Cloud Resources)
+
+permissions:
+  id-token: write # Required for requesting the JWT
+  contents: read # Required for listing the commits
+  packages: write # Required for uploading the package
+  checks: write # Required for creating a check run
+
+on:
+  # Enable manual trigger
+  workflow_dispatch:
+  schedule:
+    # Run every 4 hours on weekdays.
+    - cron: "30 0,4,8,12,16,20 * * 1-5"
+    # Run every 12 hours on weekends.
+    - cron: "30 0,12 * * 0,6"
+  # Dispatch on external events
+  repository_dispatch:
+    types: [de-functional-test]
+  push:
+    branches:
+      - main
+      - release/*
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - main
+      - features/*
+      - release/*
+
+env:
+  # Go version
+  GOVER: "1.22.5"
+  # Helm version
+  HELM_VER: "v3.15.3"
+  # KinD cluster version
+  KIND_VER: "v0.23.0"
+  # Dapr version
+  DAPR_VER: "1.12.0"
+  # Dapr dashboard version
+  DAPR_DASHBOARD_VER: "0.14.0"
+  # Kubectl version
+  KUBECTL_VER: "v1.30.0"
+
+  # The radius functional test timeout
+  FUNCTIONALTEST_TIMEOUT: 15m
+  # The base directory for storing test logs
+  RADIUS_CONTAINER_LOG_BASE: dist/container_logs
+  # The Radius helm chart location.
+  RADIUS_CHART_LOCATION: deploy/Chart/
+  # The current GitHub action link
+  ACTION_LINK: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+  # Server where terraform test modules are deployed
+  TF_RECIPE_MODULE_SERVER_URL: "http://tf-module-server.radius-test-tf-module-server.svc.cluster.local"
+  # Private Git repository where terraform module for testing is stored.
+  TF_RECIPE_PRIVATE_GIT_SOURCE: "git::https://github.com/radius-project/terraform-private-modules//kubernetes-redis"
+
+  # The number of failed tests to report.
+  ISSUE_CREATE_THRESHOLD: 2
+
+  # Local Docker registry name
+  LOCAL_REGISTRY_NAME: "radius-registry"
+  # Local Docker registry server
+  LOCAL_REGISTRY_SERVER: "localhost"
+  # Local Docker registry port
+  LOCAL_REGISTRY_PORT: "5000"
+
+jobs:
+  build:
+    name: Build Radius for test
+    runs-on: ubuntu-latest
+    env:
+      DE_IMAGE: "ghcr.io/radius-project/deployment-engine"
+      DE_TAG: "latest"
+    outputs:
+      REL_VERSION: ${{ steps.gen-id.outputs.REL_VERSION }}
+      DE_IMAGE: ${{ steps.gen-id.outputs.DE_IMAGE }}
+      DE_TAG: ${{ steps.gen-id.outputs.DE_TAG }}
```

Comment:
Where do these get set? `gen-id` runs as part of this job, so how can it provide the outputs?